src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -87,6 +87,12 @@ public static bool IsStructMarshallingRequired(TypeDesc typeDesc)
             return !IsBlittableType(type);
         }
 
+        // Set of functions that need to be resolved statically in order for 
+        // other P/Invokes to use lazy resolution.
+        // In particular, these functions are necessary to get the current application
+        // path, so having them use lazy resolution would end up causing StackOverflow
+        private static readonly string[] s_lazyPInvokeDependencies = { "GetModuleFileNameW", "CoTaskMemAlloc", "CoTaskMemFree",
```

Comment:
@luqunl Added a similar whitelist in MCG when he did a similar feature. I gave him code review feedback asking to come up with a scheme (custom attribute or something) that the class library can put on these DllImports so that MCG doesn't hardcode it.

I have no power to block MCG code reviews and don't care much about MCG hardcoding implementation details of the class library, but I do care about the CoreRT type system not to do such thing.

Please work with Luqun and Yi to come up with a way to annotate the DllImports.